### PR TITLE
Time - Corinna

### PR DESCRIPTION
**Pull request by corinna-fab**
# Task List
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe in your own words what the Model is doing in Rails | A class that adds database records based the parameters you give it as a structure.
Describe in your own words what the Controller is doing in Rails | It coordinates the actions being produced on the application.
Describe in your own words what the View is doing in Rails | Views compile the output.
Describe an edge-case controller test you wrote | Testing for what happens when an invalid id is given (IE -1)
What is the purpose of using strong params? (i.e. the params method in the controller) | To protect the attributes that users can interact with.
How are Rails migrations related to Rails models? | Rails migrations manipulate the data structure by adding columns, removing columns, etc.
Describe one area of Rails that are still unclear on | Where/how strong params can be used - IE in my "mark_complete" method where I reassign status.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+Rails.application.routes.draw do
+  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  root to: 'tasks#index'
+  resources :tasks
```
Comment (CheezItMan) [inline config/routes.rb:4]: Nice use of `resources`

File: config/routes.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+  #Leaving because of delete_path
+  delete '/tasks/:id', to: 'tasks#destroy', as: 'delete'
```
Comment (CheezItMan) [inline config/routes.rb:7]: resources makes a `destroy_task_path`, might be better to use that.

File: test/controllers/tasks_controller_test.rb [hunk 3]
```diff
@@ -88,33 +88,93 @@
...
+
+      must_respond_with :not_found
     end
   end
   
   # Complete these tests for Wave 4
   describe "destroy" do
     # Your tests go here
+    before do 
+      Task.create(name: 'Take over the world', description: 'The same thing we do everyday, Pinky.')
```
Comment (CheezItMan) [inline test/controllers/tasks_controller_test.rb:156]: Try to take over the  🌏 !

File: test/controllers/tasks_controller_test.rb [hunk 4]
```diff
@@ -88,33 +88,93 @@
...
+    end
     
+    let(:task_hash){
+      {task: {
```
Comment (CheezItMan) [inline test/controllers/tasks_controller_test.rb:160]:
```suggestion
      {
        task: {
```

File: app/controllers/tasks_controller.rb [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    elsif @task.update(task_params)
+      redirect_to task_path 
+      return
+    else 
+      render :new
+      return
+    end
+  end
+
+  def mark_complete
```
Comment (CheezItMan) [inline app/controllers/tasks_controller.rb:34]: So there's no way to mark a completed task incomplete?

File: app/controllers/tasks_controller.rb [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+  end
+
+  def destroy
+    task_id = params[:id]
+    @task = Task.find_by(id: task_id)
+
+    if @task.nil?
+      head :not_found
+      return
+    elsif
```
Comment (CheezItMan) [inline app/controllers/tasks_controller.rb:68]: elsif what?  No condition on this line.

File: app/views/tasks/edit.html.erb [hunk 7]
```diff
@@ -0,0 +1,8 @@
+<div class="todo_card">
+  <h2>Edit Page</h2>
+
+  <%= form_with model: @task, class:'task-form' do |f| %>
```
Comment (CheezItMan) [inline app/views/tasks/edit.html.erb:4]: Why do you have a `form_with` here?

The partial has the form.

File: app/views/tasks/new.html.erb [hunk 8]
```diff
@@ -0,0 +1,6 @@
+<div class="todo_card">
+<%= form_with model: @task, class:'task-form' do |f| %>
```
Comment (CheezItMan) [inline app/views/tasks/new.html.erb:2]: Similar issue here

File: app/views/tasks/show.html.erb [hunk 9]
```diff
@@ -0,0 +1,15 @@
+<div class="todo_card">
+  <h2><%= @task[:name]%></h2>
+  <br>
+  <p><%= @task[:description]%></p>
+  <% if @task[:status] == 'completed' %>
+    <h3 class="completed">Task Completed</h3>
+  <%end%>
+  <div class="button_section">
+    <%= button_to "Edit Task", edit_task_path(@task.id), :method => :get %>
```
Comment (CheezItMan) [inline app/views/tasks/show.html.erb:9]: Nice that you discovered `button_to`

**Comment by CheezItMan** [comment]
# Task List




## Major Learning Goals/Code Review





| Criteria | yes/no, and optionally any details/lines of code to reference |
| --- | --- |
| At least 6 commits with meaningful commit messages | ✔️, well done
| Routes follow RESTful conventions | ✔️, good use of resources
| Uses named routes (like `_path`) | ✔️
| Creates Models and migrations | ✔️
| Creates styled views | ✔️
| Handles errors like nonexistant tasks | ✔️
| Uses `form_with` to render forms in Rails | ✔️

## Functional Requirements/Manual Testing

| Functional Requirement | yes/no |
| --- | --- |
| Successfully handles index & show | ✔️
| index & show tests pass | ✔️
| Successfully handles: New, Create | ✔️
| New, Create tests pass | 
| Successfully handles: Edit, Update | ✔️
| Successfully handles: Destroy, Task Complete | ✔️

## Overall Feedback

| Overall Feedback | Criteria | yes/no |
| --- | --- | --- |
| Green (Meets/Exceeds Standards) | 5+ in Code Review && 5+ in Functional Requirements |



## Code Style Bonus Awards



Was the code particularly impressive in code style for any of these reasons (or more...?)

| Quality | Yes? |
| --- | --- |
| Perfect Indentation | ✅
| Elegant/Clever | ✅
| Descriptive/Readable | ✅
| Concise | ✅
| Logical/Organized | ✅


## Summary


Really nice work, I like your basic styling and you hit all the learning goals here.  The only thing I'd suggest is that your mark_complete route toggle completeness between complete and incomplete.  Otherwise outstanding work.  The tests also look quite good.
